assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js

```diff
@@ -130,32 +148,65 @@ function Media3pPane(props) {
               placeholder={__('Search', 'web-stories')}
               onSearch={onSearch}
               incremental={incrementalSearchDebounceMedia}
-              disabled={Boolean(categories.selectedCategoryId)}
+              disabled={Boolean(
+                selectedProvider &&
+                  Providers[selectedProvider].supportsCategories &&
+                  media3p[selectedProvider].categories?.selectedCategoryId
+              )}
             />
           </SearchInputContainer>
           <ProviderTabSection>
-            <ProviderTab
-              name={'Unsplash'}
-              active={true}
-              onClick={onProviderTabClick}
-            />
+            {Object.keys(Providers).map((providerType) => (
+              <ProviderTab
+                key={`provider-tab-${providerType}`}
+                name={Providers[providerType].displayName}
+                active={selectedProvider === providerType}
+                onClick={() => onProviderTabClick(providerType)}
+              />
+            ))}
           </ProviderTabSection>
-          <Media3pCategories
-            categories={categories.categories}
-            selectedCategoryId={categories.selectedCategoryId}
-            selectCategory={selectCategory}
-            deselectCategory={deselectCategory}
-          />
         </PaneHeader>
-        <PaginatedMediaGallery
-          providerType={ProviderType.UNSPLASH}
-          resources={unsplash.state.media}
-          isMediaLoading={unsplash.state.isMediaLoading}
-          isMediaLoaded={unsplash.state.isMediaLoaded}
-          hasMore={unsplash.state.hasMore}
-          setNextPage={unsplash.actions.setNextPage}
-          onInsert={insertMediaElement}
-        />
+        <PaneBottom ref={paneBottomRef}>
+          {Object.keys(Providers).map((providerType) => {
+            const wrapperProps =
+              providerType === selectedProvider
+                ? { className: 'provider-selected' }
+                : { 'aria-hidden': 'true' };
+            return (
+              <ProviderMediaCategoriesWrapper
+                dataProvider={providerType}
+                {...wrapperProps}
+                key={`provider-bottom-wrapper-${providerType}`}
+              >
+                {Providers[providerType].supportsCategories && (
+                  <Media3pCategories
+                    categories={
+                      media3p[providerType].state.categories.categories
```

Comment:
Could media3p[providerType].state / edia3p[providerType].actions be extracted to local variables?
```
<Media3pCategories
    categories={providerState.categories.categories}
    selectedCategoryId={providerState.categories.selectedCategoryId}
    selectCategory={providerActions.selectCategory}
    deselectCategory={providerActions.deselectCategory}
```

Reply:
Done.